addons/web/static/tests/core/l10n/translation_tests.js

```diff
@@ -100,6 +100,39 @@ QUnit.test("luxon is configured in the correct lang", async (assert) => {
     assert.strictEqual(DateTime.utc(2021, 12, 10).toFormat("MMMM"), "décembre");
 });
 
+QUnit.test("lang is given by an attribute on the DOM root node", async (assert) => {
+    patchWithCleanup(session.user_context, { lang: null });
+    document.documentElement.setAttribute("lang", "fr-FR");
+    registerCleanup(() => {
+        document.documentElement.removeAttribute("lang");
+    });
+    patchWithCleanup(session, {
+        cache_hashes: { translations: 1 },
+    })
+    patchWithCleanup(browser, {
+        fetch(url) {
+            assert.strictEqual(url, "/web/webclient/translations/1?lang=fr_FR");
+            return {
+                ok: true,
+                json: async () => ({
+                    modules: {},
+                    lang_parameters: {
+                        direction: "ltr",
+                        date_format: "%d/%m/%Y",
+                        time_format: "%H:%M:%S",
+                        grouping: "[3,0]",
+                        decimal_point: ",",
+                        thousands_sep: ".",
+                        week_start: 1,
+                    },
+                }),
+            };
+        },
+    });
+    serviceRegistry.add("localization", localizationService);
+    await makeTestEnv();
```

Comment:
You can use `mockRPC` instead of patching `fetch`.

```js
    serviceRegistry.add("localization", localizationService);
    await makeTestEnv({
        mockRPC(route, params) {
            assert.strictEqual(route, "/web/webclient/translations/1");
            // here asserts that lang === fr_FR (I don't remember exactly the api)
            return {
                modules: {},
                lang_parameters: {
                    direction: "ltr",
                    date_format: "%d/%m/%Y",
                    time_format: "%H:%M:%S",
                    grouping: "[3,0]",
                    decimal_point: ",",
                    thousands_sep: ".",
                    week_start: 1,
                },
            };
        },
    });
```